p2p/protocol/identify/obsaddr.go

```diff
@@ -360,26 +360,39 @@ type normalizeMultiaddrer interface {
 	NormalizeMultiaddr(addr ma.Multiaddr) ma.Multiaddr
 }
 
-func (oas *ObservedAddrManager) maybeRecordObservation(conn network.Conn, observed ma.Multiaddr) {
+type addrsProvider interface {
+	Addrs() []ma.Multiaddr
+}
+type listenAddrsProvider interface {
+	ListenAddresses() []ma.Multiaddr
+	InterfaceListenAddresses() ([]ma.Multiaddr, error)
+}
+
+type connMultiaddrProvider interface {
+	RemoteMultiaddr() ma.Multiaddr
+	LocalMultiaddr() ma.Multiaddr
+}
```

Comment:
Reuse this interface? https://github.com/libp2p/go-libp2p/blob/952f7cbc5cf5d3e16d9171f1151b20c2a227d484/core/network/conn.go#L70-L80